internal/command/deploy/plan.go

```diff
@@ -352,6 +359,13 @@ func (md *machineDeployment) updateProcessGroup(ctx context.Context, machineTupl
 
 			sl := machineLogger.getLoggerFromID(machineID)
 
+			select {
+			case <-gCtx.Done():
+				sl.LogStatus(statuslogger.StatusFailure, "skipping machine update due to earlier failure")
+				return gCtx.Err()
+			default:
+			}
```

Comment:
You can just check `if err := gCtx.Err(); err != nil { ... }` directly instead of doing the `select` block.